src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombiner.java

```diff
@@ -166,18 +167,35 @@ private Map<Integer, Object[]> getDocIdSortFieldsMap(
         Map<Integer, Object[]> docIdSortFieldMap = new HashMap<>();
         final List<TopDocs> topFieldDocs = compoundTopDocs.getTopDocs();
         final boolean isSortByScore = isSortOrderByScore(sort);
+        final boolean isCollapseEnabled = topFieldDocs.getFirst() instanceof CollapseTopFieldDocs;
         for (TopDocs topDocs : topFieldDocs) {
+            int i = 0;
             for (ScoreDoc scoreDoc : topDocs.scoreDocs) {
                 FieldDoc fieldDoc = (FieldDoc) scoreDoc;
 
                 if (docIdSortFieldMap.get(fieldDoc.doc) == null) {
                     // If sort by score then replace sort field value with normalized score.
                     if (isSortByScore) {
-                        docIdSortFieldMap.put(fieldDoc.doc, new Object[] { combinedNormalizedScoresByDocId.get(fieldDoc.doc) });
+                        docIdSortFieldMap.put(
+                            fieldDoc.doc,
+                            isCollapseEnabled
+                                ? new Object[] {
+                                    combinedNormalizedScoresByDocId.get(fieldDoc.doc),
+                                    ((CollapseTopFieldDocs) topDocs).collapseValues[i] }
+                                : new Object[] { combinedNormalizedScoresByDocId.get(fieldDoc.doc) }
+                        );
                     } else {
-                        docIdSortFieldMap.put(fieldDoc.doc, fieldDoc.fields);
+                        if (isCollapseEnabled) {
+                            Object[] fields = new Object[fieldDoc.fields.length + 1];
+                            System.arraycopy(fieldDoc.fields, 0, fields, 0, fieldDoc.fields.length);
```

Comment:
why we need a copy of fields array?

Reply:
We are using the fields object of the FieldDoc to also pass the collapse value.  Therefore, we need to copy the old fields array to a new array and append the collapse value as the last element in the new array